src/amp-story-player/amp-story-player-iframe-pool.js

```diff
@@ -77,4 +77,49 @@ export class IframePool {
 
     return detachedStoryIdx;
   }
+
+  /**
+   * Clears stored story indices.
+   * @return {!Array<number>}
+   */
+  evictStories() {
+    const evictedStories = this.storyIdsWithIframe_;
+    this.storyIdsWithIframe_ = [];
+    return evictedStories;
+  }
+
+  /**
+   * @return {!Array<number>}
+   */
+  getAvailableIframeIdx() {
+    return this.iframePool_;
+  }
+
+  /**
+   * Finds adjacent iframe indices given an index.
+   * Examples of resulting adjacent arrays:
+   * 0 -> [0] [1] [2] [] []
+   * 1 -> [0] [1] [2] [] []
+   * 4 -> [] [] [0] [1] [2]
+   * 1 -> [0] [1]
+   * @param {number} storyIdx
+   * @param {number} maxIdx
+   * @return {!Array<number>}
+   */
+  findAdjacent(storyIdx, maxIdx) {
+    const adjacent = [];
+
+    // If index is at rightmost part of the array, adjacent iframes will all be
+    // at the left.
+    let cursor = storyIdx + 1 > maxIdx ? storyIdx - 2 : storyIdx - 1;
```

Comment:
Do we really want to preload the previous 2 stories? I think loading +1/-1 is already a lot (3 stories), if in this case we could only load 2 stories that's better?

Reply:
Since we assume we have the same number of iframes when we build the player and as we navigate the stories, this would require some changes in how we handle navigation (evict n-2 iframe and assign it to n+1, for e.g.). Can we leave this as an optimization follow up in the issue I filed?

Reply:
Sounds good, but let's make sure we work on these, it's very important to get the performance right here if we want to drive adoption :))